protocols/bftcosi/bftcosi.go

```diff
@@ -155,43 +151,59 @@ func (bft *ProtocolBFTCoSi) Start() error {
 	if err := bft.startAnnouncement(RoundPrepare); err != nil {
 		return err
 	}
-	return bft.startAnnouncement(RoundCommit)
+	go func() {
+		//time.Sleep(time.Second)
+		bft.startAnnouncement(RoundCommit)
+	}()
+	return nil
 }
 
-// Dispatch listens on all channels and implements the sda.ProtocolInstance
-// interface.
+// Dispatch makes sure that the order of the messages is correct by waiting
+// on each channel in the correct order.
+// By closing the channels for the leafs we can avoid having
+// `if !bft.IsLeaf` in the code.
 func (bft *ProtocolBFTCoSi) Dispatch() error {
-	log.Lvl2(bft.Name(), "Starts")
-	for {
-		var err error
-		select {
-		case msg := <-bft.announceChan:
-			// Announcement
-			err = bft.handleAnnouncement(msg.Announce)
-		case msg := <-bft.commitChan:
-			// Commitment
-			err = bft.handleCommitment(msg.Commitment)
-
-		case msg := <-bft.challengePrepareChan:
-			// Challenge
-			err = bft.handleChallengePrepare(&msg.ChallengePrepare)
-
-		case msg := <-bft.challengeCommitChan:
-			err = bft.handleChallengeCommit(&msg.ChallengeCommit)
-
-		case msg := <-bft.responseChan:
-			// Response
-			err = bft.startResponse(msg.Response.TYPE, &msg.Response)
-		case <-bft.doneProcessing:
-			// we are done
-			log.Lvl2(bft.Name(), "BFTCoSi Dispatches stop.")
+	bft.closingMutex.Lock()
+	if bft.closing {
+		return nil
+	}
+	// Close unused channels for the leaf nodes. This is not easily
+	// possible for the root-node.
```

Comment:
What does it mean `not easily possible for the root-node` ?
BTW, Golang takes care of that alone, we don't have to close necessarily a channel just because *it is not used anymore* (i.e. it's not like a file descriptor).
So do we need that ?